[macOS] Update Info window.

nevack opened the pull request:
Use SF Symbols on macOS 11+

Use NSSegmentedControl instead of custom view.

Add NSImage extension to fallback loading from
assets on macOS 10.15 and older.

Also should fix sizing issues when changing pane from Options
to Files.

Fixes #2247

Result:
<img src="https://user-images.githubusercontent.com/8330119/145144659-24e3f61d-3219-4167-abe3-ae4410903fe4.png" width="600" />

GScottElshaw commented:
Killing me with expectation...

nevack commented on macosx/InfoWindowController.mm:
```diff
@@ -91,12 +91,31 @@ - (void)awakeFromNib
+                             withFallback:@"InfoFiles"]
+         forSegment:TAB_FILE_TAG];
+    [fTabs setImage:[NSImage systemSymbol:@"gearshape"
+                             withFallback:@"InfoIotions"]
```
oops, typo

nevack commented on macosx/InfoWindow.xib:
```diff
@@ -12,151 +12,109 @@
-                        </connections>
-                    </matrix>
+                            <segments>
+                                <segment image="info.circle" catalog="system" selected="YES"/>
```
I think I can adjust it to make this part uniform

livings124 commented on macosx/en.lproj/InfoOptionsView.xib:
```diff
@@ -1,8 +1,8 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.Cocoa.XIB" version="3.0" toolsVersion="11762" systemVersion="16D32" targetRuntime="MacOSX.Cocoa" propertyAccessControl="none" useAutolayout="YES">
```
What'd you change in this file? If we change something here, we have to do it in all locales unfortunately.

livings124 commented on macosx/InfoWindowController.mm:
```diff
@@ -91,12 +91,31 @@ - (void)awakeFromNib
+    [fTabs setImage:[NSImage systemSymbol:@"antenna.radiowaves.left.and.right"
+                             withFallback:@"InfoTracker"]
+         forSegment:TAB_TRACKERS_TAG];
+    [fTabs setImage:[NSImage systemSymbol:@"person.fill"
```
I'd not use the fill version here, since that doesn't match the others which are outlines

livings124 replied:
Also, the same symbol as the one used in Peers preferences might make sense here, since they're the same name.

livings124 commented on macosx/InfoWindowController.mm:
```diff
@@ -91,12 +91,31 @@ - (void)awakeFromNib
+    [fTabs setImage:[NSImage systemSymbol:@"number.square"
+                             withFallback:@"InfoActivity"]
+         forSegment:TAB_ACTIVITY_TAG];
+    [fTabs setImage:[NSImage systemSymbol:@"antenna.radiowaves.left.and.right"
```
I don't love that this is the same icon as Settings > Remote. Any thoughts on a different one either for this or that?

nevack commented:
Hi, @livings124. Can you take a look?

livings124 reviewed (commented):
This is looking great. My last concern is just the server rack for activity, since I think the icon itself is a bit unclear what it is and BitTorrent is by design not server-based. Maybe the previous icon what fine there after all?

livings124 commented:
> > server rack for activity
> 
> It's not for activity, it's for Tracker tab
> 
> 

Eep, my mistake. But I think the reinforces that it's a bit unclear. I'll give a look through the symbols, but the previous antenna might be best after all.

nevack commented:
> > > server rack for activity
> > 
> > 
> > It's not for activity, it's for Tracker tab
> 
> Eep, my mistake. But I think the reinforces that it's a bit unclear. I'll give a look through the symbols, but the previous antenna might be best after all.

Want to note that it's important to look symbol availability, some of them are available only on macOS 12/ iOS 15

mangosplat commented:
Suggestion: perhaps chart.pie or arrow.right.arrow.left.square would make for a better representation of the 'Activity' tab (visually shows transfer progress so I think the pie chart makes sense; arrows to represent transfer)

![image](https://user-images.githubusercontent.com/95374372/145660726-63683855-df45-470e-94b7-04930c809704.png)

![image](https://user-images.githubusercontent.com/95374372/145660748-3420ba6f-3d99-411a-b662-702507df0faf.png)

GScottElshaw commented:
> > > > server rack for activity
> > > 
> > > 
> > > It's not for activity, it's for Tracker tab
> > 
> > 
> > Eep, my mistake. But I think the reinforces that it's a bit unclear. I'll give a look through the symbols, but the previous antenna might be best after all.
> 
> Want to note that it's important to look symbol availability, some of them are available only on macOS 12/ iOS 15

My 2 cents is that everything looks great! My preference is for the 'Activity' icon to be a jigsaw piece rather than a hash symbol, mostly because, unless there was a chessboard icon, the information reflects parts of a whole.

nevack commented:
@livings124
 - Changed back Tracker tab icon.
 - Updated Activity tab icon.
 - Update FavIcon in Trackers list with SF Symbol.

![Info](https://user-images.githubusercontent.com/8330119/145701574-c4d992f1-b770-4445-83fd-70a975723704.png)

livings124 commented:
Thanks- looks good!

GScottElshaw commented:
Oops. MacOS 11.6.1 | 3.00+ (2b3fbd93f5)

![Screenshot 2021-12-13 at 11 13 46 AM](https://user-images.githubusercontent.com/69029666/145731630-59e91aa2-0fc7-452a-aed1-cc894c677fdf.png)

nevack commented:
> Oops. MacOS 11.6.1 | 3.00+ ([2b3fbd9](https://github.com/transmission/transmission/commit/2b3fbd93f5aec08879759bed6e8fe36dcf19f1f3)) @nevack @livings124
> 
> ![Screenshot 2021-12-13 at 11 13 46 AM](https://user-images.githubusercontent.com/69029666/145731630-59e91aa2-0fc7-452a-aed1-cc894c677fdf.png)

My bad, I've made the mistake that I tried to prevent others from.

<img width="305" alt="image" src="https://user-images.githubusercontent.com/8330119/145752456-6bf6e07e-c647-44f3-942c-f5932f7e487f.png">

I'll make fix ASAP

mikedld commented:
@nevack, could you please look into this, sounds related: https://forum.transmissionbt.com/viewtopic.php?t=20925&p=80299

nevack commented:
> @nevack, could you please look into this, sounds related: https://forum.transmissionbt.com/viewtopic.php?t=20925&p=80299

Hmm, maybe some bug on older macOS versions.
Need more info about environment, then I can debug it in VM

livings124 commented:
> > @nevack, could you please look into this, sounds related: https://forum.transmissionbt.com/viewtopic.php?t=20925&p=80299
> 
> Hmm, maybe some bug on older macOS versions. Need more info about environment, then I can debug it in VM

Sounds like it's macOS 10.14.6

beren12 commented:
Yes, it is a bug on older os x, what info do you need? it happens on 10.13.

livings124 commented:
> Yes, it is a bug on older os x, what info do you need? it happens on 10.13.

If you open up Console.app, when you try to change sections does it print anything?

beren12 commented:
> Yes, it is a bug on older os x, what info do you need? it happens on 10.13.
> 
> If you open up Console.app, when you try to change sections does it print anything?
> 

No, only this when opening the app:

2022-02-20 17:04:49.735 Transmission[86267:21305939] Could not find image named 'info.circle'.
2022-02-20 17:04:49.735 Transmission[86267:21305939] Could not find image named 'square.grid.3x3.fill.square'.
2022-02-20 17:04:49.736 Transmission[86267:21305939] Could not find image named 'antenna.radiowaves.left.and.right'.
2022-02-20 17:04:49.737 Transmission[86267:21305939] Could not find image named 'person.2'.
2022-02-20 17:04:49.738 Transmission[86267:21305939] Could not find image named 'doc.on.doc'.
2022-02-20 17:04:49.738 Transmission[86267:21305939] Could not find image named 'gearshape'.

livings124 commented:
@beren12 I merged in https://github.com/transmission/transmission/pull/2665 which should remove those warnings. Could you check to see if the issue remains?

beren12 commented:
The warning is gone, the info pane doesn't change.

beren12 commented:
still the same as of https://build.transmissionbt.com/job/trunk-mac/7738/

RobFog commented:
There aren't many issues left on the Transmission 4 beta milestone. #2599 is one of them. @nevack, are you still interested in fixing this?

nevack commented:
> There aren't many issues left on the Transmission 4 beta milestone. #2599 is one of them. @nevack, are you still interested in fixing this?

Hi, I have set up VM with 10.14.6 and reproduced the issue.
Quite interesting, `[InfoWindowController setTab:]` is called, while `[NSControl selectedTag]` always returns zero.

I'll make PR as soon as solution is found

sweetppro commented:
I actually was also just looking at this, and submitted a fix:
https://github.com/transmission/transmission/pull/2869

nevack commented:
> I actually was also just looking at this, and submitted a fix: #2869

Damn, https://github.com/transmission/transmission/pull/2870 😄

sweetppro commented:
> > I actually was also just looking at this, and submitted a fix: #2869
> 
> Damn, #2870 😄

ha nice!
Ill remove mine.
your PR fits the coding style better I think
